### PR TITLE
chore(typing): avoid runtime errors in Python 3.5

### PR DESCRIPTION
**Pull request by gnufede**
`typing.DefaultDict` is not available on Python 3.5, so this PR avoids runtime errors while keeping this import for type checking.

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) are followed.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Author is aware of the performance implications of this PR as reported in the benchmarks PR comment.

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer is aware of, and discussed the performance implications of this PR as reported in the benchmarks PR comment.

### CODE REVIEW

File: ddtrace/internal/module.py [hunk 1]
```diff
@@ -27,8 +34,6 @@
 log = get_logger(__name__)
 
 ModuleHookType = Callable[[ModuleType], None]
```
Comment (brettlangdon) [inline ddtrace/internal/module.py:36]: does this need to move too?

Comment (gnufede) [reply ddtrace/internal/module.py:36]: We import it at ``ddtrace.debugging._debugger``, so it cannot be under the if (unless we change it there too)

Comment (brettlangdon) [reply ddtrace/internal/module.py:36]: oh, I see we are keeping `ModuleType`, and `Callable` not in the if statement 👍🏻

**Review by emmettbutler — APPROVED** [review]

**Review by mabdinur — COMMENTED** [review]
We import DefaultDict in `ddtrace._hooks.py`, `ddtrace.internal.logger.py`, and `ddtrace.internal.processor.trace.py`. Should we add `if typing.TYPE_CHECKING` check on those imports as well?

**Review by ZStriker19 — APPROVED** [review]

File: ddtrace/internal/module.py [hunk 2]
```diff
@@ -6,19 +6,26 @@
 from os.path import join
 import sys
 from types import ModuleType
-from typing import Any
+import typing
 from typing import Callable
```
Comment (mabdinur) [inline ddtrace/internal/module.py:10]:
```suggestion
```

File: ddtrace/internal/module.py [hunk 3]
```diff
@@ -6,19 +6,26 @@
...
-from typing import Optional
-from typing import Set
-from typing import Tuple
-from typing import Type
-from typing import Union
 from typing import cast
 
+
+if typing.TYPE_CHECKING:
+    from typing import Any
```
Comment (mabdinur) [inline ddtrace/internal/module.py:15]:
```suggestion
    from typing import Any
    from typing import Callabe
```

File: ddtrace/internal/module.py [hunk 4]
```diff
@@ -6,19 +6,26 @@
 from os.path import join
 import sys
 from types import ModuleType
```
Comment (mabdinur) [inline ddtrace/internal/module.py:8]:
```suggestion
```

File: ddtrace/internal/module.py [hunk 5]
```diff
@@ -6,19 +6,26 @@
...
-from typing import DefaultDict
-from typing import Dict
-from typing import Iterator
-from typing import List
-from typing import Optional
-from typing import Set
-from typing import Tuple
-from typing import Type
-from typing import Union
 from typing import cast
```
Comment (mabdinur) [inline ddtrace/internal/module.py:11]:
```suggestion
```

File: ddtrace/internal/module.py [hunk 6]
```diff
@@ -6,19 +6,26 @@
...
+    from typing import DefaultDict
+    from typing import Dict
+    from typing import Iterator
+    from typing import List
+    from typing import Optional
+    from typing import Set
+    from typing import Tuple
+    from typing import Type
+    from typing import Union
+
```
Comment (mabdinur) [inline ddtrace/internal/module.py:25]:
```suggestion

    ModuleHookType = Callable[[ModuleType], None]
```

File: ddtrace/internal/module.py (hunk 1 above)
Comment (mabdinur) [inline ddtrace/internal/module.py:36]:
```suggestion
```

**Review by mabdinur — COMMENTED** [review]
nit: import typing.ModuleType and typing.Callable if typing.TYPE_CHECKING is True 

Note - typing.cast is required by `ModuleWatchdog.find_spec(...)` and `ModuleWatchdog.unregister_origin_hook(..)`.

**Review by mabdinur — APPROVED** [review]

**Pull request merged**